### ISSUE TITLE
Failed to install docker on Ubuntu - iptables error

### ISSUE THREAD

**Issue by sbenedetto** [moby/moby#32349]
I'm trying to install docker on ubuntu 16.04

Kernel is not the default, I'm using 4.11.0-rc3, so at first I was missing some modules but I managed to fix that.

Now `systemctl restart docker.service` fails with the following iptables error

```
Apr  4 14:50:32 localhost dockerd[4241]: time="2017-04-04T14:50:32.866592972+01:00" level=warning msg="mountpoint for pids not found"
Apr  4 14:50:32 localhost dockerd[4241]: time="2017-04-04T14:50:32.867241786+01:00" level=info msg="Loading containers: start."
Apr  4 14:50:32 localhost dockerd[4241]: time="2017-04-04T14:50:32.893893122+01:00" level=info msg="Firewalld running: false"
Apr  4 14:50:33 localhost dockerd[4241]: time="2017-04-04T14:50:33.180044219+01:00" level=info msg="Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon option --bip can be used to set a preferred IP address"
Apr  4 14:50:33 localhost dockerd[4241]: Error starting daemon: Error initializing network controller: Error creating default "bridge" network: Failed to program NAT chain: Failed to inject docker in PREROUTING chain: iptables failed: **iptables --wait -t nat -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER: iptables**: No chain/target/match by that name.
Apr  4 14:50:33 localhost dockerd[4241]:  (exit status 1)
```
This is the content of my nat table

```
iptables -t nat --list
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination         

Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination         
MASQUERADE  all  --  172.17.0.0/16        anywhere            
RETURN     all  --  192.168.122.0/24     base-address.mcast.net/24 
RETURN     all  --  192.168.122.0/24     255.255.255.255     
MASQUERADE  tcp  --  192.168.122.0/24    !192.168.122.0/24     masq ports: 1024-65535
MASQUERADE  udp  --  192.168.122.0/24    !192.168.122.0/24     masq ports: 1024-65535
MASQUERADE  all  --  192.168.122.0/24    !192.168.122.0/24    

Chain DOCKER (0 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             anywhere            
```

This are the modules currently loaded

```
xfrm_user              49152  0
xfrm_algo              20480  1 xfrm_user
br_netfilter           36864  0
ebtable_filter         16384  0
```

I tried to delete /var/lib/docker and reinstall (after fixing some missing modules) but nothing.

**Comment by thaJeztah** [comment]
This looks like a duplicate of https://github.com/docker/docker/issues/25393, and https://github.com/docker/docker/issues/16816. The first one should be resolved in docker 17.04, which will be released this week

**Comment by sbenedetto** [comment]
Here is what I did

apt remove docker-engine
rm -rf /var/lib/docker
iptables -F DOCKER
iptables -X DOCKER
iptables -F DOCKER-ISOLATION
iptables -X DOCKER-ISOLATION

to make sure was in a clear state. Then I run

curl -fsSL https://get.docker.com/ | sh

to install docker 17.04.0~ce

Same issue.

On a different note, I was able to install 17.04.0~ce on Fedora 24 with a custom kernel 4.10.8 just fine.
